app/src/main/java/com/allan/boardbuddies/fragments/EditBoardFragment.java

```diff
@@ -0,0 +1,114 @@
+    private CanvasView canvasView;
+    private Toolbar toolbar;
+    private BottomNavigationView bottomNavigationView;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_edit_board, container, false);
+        toolbar = view.findViewById(R.id.board_toolbar);
+        boardTitle = view.findViewById(R.id.board_title);
+        canvasView = view.findViewById(R.id.canvas_view);
+        bottomNavigationView = view.findViewById(R.id.board_bottom_nav);
+
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        editBoardViewModel = new ViewModelProvider(requireActivity()).get(EditBoardViewModel.class);
+        editBoardViewModel.getTitle().observe(getViewLifecycleOwner(), title -> boardTitle.setText(title));
+
+        ViewTreeObserver viewTreeObserver = canvasView.getViewTreeObserver(); // gaurentees view is laid out vs post runnable
+        if (viewTreeObserver.isAlive()) {
+            viewTreeObserver.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+                @Override
+                public void onGlobalLayout() {
+                    canvasView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                    canvasView.init(canvasView.getWidth(), canvasView.getHeight());
+                }
+            });
+        }
+
+        toolbar.setNavigationOnClickListener(v -> {
+            editBoardViewModel.saveBoard(boardTitle.getText().toString(), canvasView.getStrokes(), canvasView.getTextBoxes());
+            Navigation.findNavController(requireActivity(), R.id.nav_host).popBackStack();
+        });
+
+        if (savedInstanceState == null) {
+            editBoardViewModel.setBoard(getArguments().getInt("position", -1));
+        }
+        boardTitle.setText(editBoardViewModel.getTitle().getValue());
+        canvasView.setStrokes(editBoardViewModel.getStrokes().getValue());
+        canvasView.setTexts(editBoardViewModel.getTexts().getValue());
+
+        bottomNavigationView.setOnItemSelectedListener(this);
+
+        getChildFragmentManager().setFragmentResultListener(Constants.TEXT_DIALOG_FRAGMENT_ADD_TEXT_REQUEST_KEY, this, new FragmentResultListener() {
+            @Override
+            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle bundle) {
+                String result = bundle.getString(Constants.TEXT_DIALOG_FRAGMENT_ADDED_TEXT);
+                if (!result.isEmpty()){
+                    canvasView.newTextBox(result);
+                }
+            }
+        });
+    }
+
+    @Override
+    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()){
+            case R.id.add_text:
```

Comment:
remind me again how you would go about adding a text box? don't you just click anywhere on the board and not the nav bar?

my understanding is that this callback would only handle the case where you click the nav bar, right?

Reply:
No you get a new textbox if you click the navbar button